_targets.R

```diff
@@ -0,0 +1,150 @@
+library(targets)
+library(tarchetypes)
+library(reticulate)
+
+yaml_file = 'test_config.yml'
+
+# MUST READ ---------------------------------------------------------------
+
+# IMPORTANT NOTE:
+#
+# you must execute the command 'earthengine authenticate' in a zsh terminal 
+# before initializing this workflow. See the repository README for complete 
+# dependencies and troubleshooting.
+
+# RUNNING {TARGETS}:
+#
+# run tar_make() to run the pipeline
+# run tar_visnetwork() to see visual summary
+
+
+# Set up python virtual environment ---------------------------------------
+
+if(!dir.exists('env')) {
+  source('data_acquisition/src/pySetup.R')
+} else {
+  use_condaenv(file.path(getwd(), 'env'))
+}
+
+
+# Source functions --------------------------------------------------------
+
+source('data_acquisition/src/general_functions.R')
+source_python('data_acquisition/src/gee_functions.py')
+
+# Define {targets} workflow -----------------------------------------------
+
+# Set target-specific options such as packages.
+tar_option_set(packages = "tidyverse")
+
+# target objects in workflow
+list(
```

Comment:
As you've suggested, I *think* that the yml processing could be simplified. In the aquasat workflow I have one target dedicated to reading it in with `tar_file_read()` and no hard link to the file before the pipeline list. For example:
```
tar_file_read(
    config_yml,
    "test_config.yml",
    read = read_yaml(!!.x),
    packages = "yaml"
  )
```

Reply:
I was trying to make it a little bit more 'user friendly' by allowing the user to call a specific config file - I'm worried that burying the config file in the targets workflow (mostly metaphorically, I know it's only 70 lines down) will not make it very inviting. In an ideal world, users don't have to necessarily move files into the repo, they can just point to them in the config file. I'd be interested in chatting with you about the philosophy of one over the other - especially if the added context is that the end goal of this repository is more a 'template repo' than a repo that is pulling for a specific ROSS project.